lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -7701,6 +7701,19 @@ InferredGenericSignatureRequest::evaluate(
   // from that context.
   builder.addGenericSignature(parentSig);
 
+  // Type check the generic parameters, treating all generic type
+  // parameters as dependent, unresolved.
+  SmallVector<GenericParamList *, 2> gpLists;
+  if (gpl->getOuterParameters() && !parentSig) {
```

Comment:
getOuterParameters() is only used for extensions so parentSig should always be nullptr. You could assert instead